File: problem_3.rb
require 'prime'

def prime_factors(n)
  prime_factors = []
  divisor = 2
  while n > 1
    while n % divisor == 0
      n /= divisor
      prime_factors << divisor
    end
    divisor += 1
  end
  prime_factors.last
end

p prime_factors(600851475143)

# => 6857

File: problem_3.js
var primeFactors = function(n) {
  var prime_factors = [];
  var divisor = 2;

  while (n > 1) {
    if (n % divisor === 0) {
      n = n / divisor;
      prime_factors.push(divisor);
    }
    divisor = divisor + 1;
  }
  console.log(prime_factors[prime_factors.length - 1]);
}

primeFactors(600851475143);

// => 6857
